### ISSUE TITLE
How would I upgrade my version of this package?

### ISSUE THREAD

**Issue by Frick-David** [DAINRA/ungoogled-chromium-void#3]
I am building from source.

When I run
```
xbps-install -vR hostdir/binpkgs ungoogled-chromium
```

I get a message that it is already installed.
Thanks for the feedback.

**Comment by KF-Art** [comment]
Greetings. 
Did you tried with the _-f_ (force) parameter? It will not allow you to install a package if the version is the same, so its necessary to force the installation.

**Comment by DAINRA** [comment]
Yeah, use the -f parameter, that's realy the only way besides version bumping or uninstalling and reinstalling.

**Comment by DAINRA** [comment]
Yes it will keep them, those are user files (usually found in `~/.config/chromium`) and xbps-install only deals with the files in the package (you can read more about it [here](https://docs.voidlinux.org/xbps/index.html)).
That being said, if you are using any custom flags in your `/usr/bin/ungoogled-chromium` (for example `--enable-features=VaapiVideoDecoder`) it might be a good idea to add those flags to the template:
https://github.com/DAINRA/ungoogled-chromium-void/blob/7840292e93a3cc9c4da8e52ef92c8c97fb6a4521/ungoogled-chromium/files/chromium.sh#L4
That file does get replaced on install/reinstall/update.

AFAIK the only files touched are:
```
/usr/bin/ungoogled-chromium
/usr/lib/ungoogled-chromium/chrome_100_percent.pak
/usr/lib/ungoogled-chromium/chrome_200_percent.pak
/usr/lib/ungoogled-chromium/chrome_crashpad_handler
/usr/lib/ungoogled-chromium/chrome_sandbox
/usr/lib/ungoogled-chromium/chromedriver
/usr/lib/ungoogled-chromium/headless_lib.pak
/usr/lib/ungoogled-chromium/icudtl.dat
/usr/lib/ungoogled-chromium/libEGL.so
/usr/lib/ungoogled-chromium/libGLESv2.so
/usr/lib/ungoogled-chromium/locales/am.pak
/usr/lib/ungoogled-chromium/locales/ar.pak
/usr/lib/ungoogled-chromium/locales/bg.pak
/usr/lib/ungoogled-chromium/locales/bn.pak
/usr/lib/ungoogled-chromium/locales/ca.pak
/usr/lib/ungoogled-chromium/locales/cs.pak
/usr/lib/ungoogled-chromium/locales/da.pak
/usr/lib/ungoogled-chromium/locales/de.pak
/usr/lib/ungoogled-chromium/locales/el.pak
/usr/lib/ungoogled-chromium/locales/en-GB.pak
/usr/lib/ungoogled-chromium/locales/en-US.pak
/usr/lib/ungoogled-chromium/locales/es-419.pak
/usr/lib/ungoogled-chromium/locales/es.pak
/usr/lib/ungoogled-chromium/locales/et.pak
/usr/lib/ungoogled-chromium/locales/fa.pak
/usr/lib/ungoogled-chromium/locales/fi.pak
/usr/lib/ungoogled-chromium/locales/fil.pak
/usr/lib/ungoogled-chromium/locales/fr.pak
/usr/lib/ungoogled-chromium/locales/gu.pak
/usr/lib/ungoogled-chromium/locales/he.pak
/usr/lib/ungoogled-chromium/locales/hi.pak
/usr/lib/ungoogled-chromium/locales/hr.pak
/usr/lib/ungoogled-chromium/locales/hu.pak
/usr/lib/ungoogled-chromium/locales/id.pak
/usr/lib/ungoogled-chromium/locales/it.pak
/usr/lib/ungoogled-chromium/locales/ja.pak
/usr/lib/ungoogled-chromium/locales/kn.pak
/usr/lib/ungoogled-chromium/locales/ko.pak
/usr/lib/ungoogled-chromium/locales/lt.pak
/usr/lib/ungoogled-chromium/locales/lv.pak
/usr/lib/ungoogled-chromium/locales/ml.pak
/usr/lib/ungoogled-chromium/locales/mr.pak
/usr/lib/ungoogled-chromium/locales/ms.pak
/usr/lib/ungoogled-chromium/locales/nb.pak
/usr/lib/ungoogled-chromium/locales/nl.pak
/usr/lib/ungoogled-chromium/locales/pl.pak
/usr/lib/ungoogled-chromium/locales/pt-BR.pak
/usr/lib/ungoogled-chromium/locales/pt-PT.pak
/usr/lib/ungoogled-chromium/locales/ro.pak
/usr/lib/ungoogled-chromium/locales/ru.pak
/usr/lib/ungoogled-chromium/locales/sk.pak
/usr/lib/ungoogled-chromium/locales/sl.pak
/usr/lib/ungoogled-chromium/locales/sr.pak
/usr/lib/ungoogled-chromium/locales/sv.pak
/usr/lib/ungoogled-chromium/locales/sw.pak
/usr/lib/ungoogled-chromium/locales/ta.pak
/usr/lib/ungoogled-chromium/locales/te.pak
/usr/lib/ungoogled-chromium/locales/th.pak
/usr/lib/ungoogled-chromium/locales/tr.pak
/usr/lib/ungoogled-chromium/locales/uk.pak
/usr/lib/ungoogled-chromium/locales/vi.pak
/usr/lib/ungoogled-chromium/locales/zh-CN.pak
/usr/lib/ungoogled-chromium/locales/zh-TW.pak
/usr/lib/ungoogled-chromium/resources.pak
/usr/lib/ungoogled-chromium/snapshot_blob.bin
/usr/lib/ungoogled-chromium/swiftshader/libEGL.so
/usr/lib/ungoogled-chromium/swiftshader/libGLESv2.so
/usr/lib/ungoogled-chromium/ungoogled-chromium
/usr/lib/ungoogled-chromium/v8_context_snapshot.bin
/usr/share/applications/ungoogled-chromium.desktop
/usr/share/icons/hicolor/128x128/apps/ungoogled-chromium.png
/usr/share/icons/hicolor/16x16/apps/ungoogled-chromium.png
/usr/share/icons/hicolor/24x24/apps/ungoogled-chromium.png
/usr/share/icons/hicolor/256x256/apps/ungoogled-chromium.png
/usr/share/icons/hicolor/32x32/apps/ungoogled-chromium.png
/usr/share/icons/hicolor/48x48/apps/ungoogled-chromium.png
/usr/share/icons/hicolor/64x64/apps/ungoogled-chromium.png
/usr/share/licenses/ungoogled-chromium/LICENSE
/usr/bin/chromedriver -> /usr/lib/ungoogled-chromium/chromedriver
```

**Comment by Frick-David** [comment]
@DAINRA with the most recent release. I get a build error. I was wondering if you could give me some pointers to solve this:

```
=> ungoogled-chromium-95.0.4638.54_1: running do_build ...
ninja: Entering directory `out/Release'
[1/26465] ACTION //base/third_party/libevent:gen_libevent_shim(//build/toolchain/linux/unbundle:default)
[2/25497] ACTION //third_party/ffmpeg:gen_ffmpeg_shim(//build/toolchain/linux/unbundle:default)
[3/25492] ACTION //third_party/flac:gen_flac_shim(//build/toolchain/linux/unbundle:default)
[4/25490] ACTION //third_party/libpng:gen_libpng_shim(//build/toolchain/linux/unbundle:default)
[5/25488] ACTION //third_party/libwebp:gen_libwebp_shim(//build/toolchain/linux/unbundle:default)
[6/25381] CXX obj/components/paint_preview/common/common/subset_font.o
FAILED: obj/components/paint_preview/common/common/subset_font.o 
clang++ -MMD -MF obj/components/paint_preview/common/common/subset_font.o.d -DUSE_UDEV -DUSE_AURA=1 -DUSE_GLIB=1 -DUSE_NSS_CERTS=1 -DUSE_OZONE=1 -DUSE_X11=1 -DOFFICIAL_BUILD -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -DNO_UNWIND_TABLES -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D_FORTIFY_SOURCE=2 -DNDEBUG -DNVALGRIND -DDYNAMIC_ANNOTATIONS_ENABLED=0 -DGLIB_VERSION_MAX_ALLOWED=GLIB_VERSION_2_40 -DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_40 -DU_USING_ICU_NAMESPACE=0 -DU_ENABLE_DYLOAD=0 -DUSE_CHROMIUM_ICU=1 -DU_ENABLE_TRACING=1 -DU_ENABLE_RESOURCE_TRACING=0 -DU_STATIC_IMPLEMENTATION -DICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_FILE -DSK_UNTIL_CRBUG_1187654_IS_FIXED -DSK_CODEC_DECODES_PNG -DSK_CODEC_DECODES_WEBP -DSK_ENCODE_PNG -DSK_ENCODE_WEBP -DSK_ENABLE_SKSL -DSK_USER_CONFIG_HEADER=\"../../skia/config/SkUserConfig.h\" -DSK_GL -DSK_CODEC_DECODES_JPEG -DSK_ENCODE_JPEG -DSK_HAS_WUFFS_LIBRARY -DSK_VULKAN_HEADER=\"../../skia/config/SkVulkanConfig.h\" -DSK_VULKAN=1 -DSK_SUPPORT_GPU=1 -DSK_GPU_WORKAROUNDS_HEADER=\"gpu/config/gpu_driver_bug_workaround_autogen.h\" -DVK_USE_PLATFORM_XCB_KHR -DGOOGLE_PROTOBUF_NO_RTTI -DGOOGLE_PROTOBUF_NO_STATIC_INITIALIZER -DHAVE_PTHREAD -I../.. -Igen -I../../third_party/perfetto/include -Igen/third_party/perfetto/build_config -Igen/third_party/perfetto -Igen/shim_headers/libevent_shim -Igen/shim_headers/libpng_shim -Igen/shim_headers/libwebp_shim -I../../third_party/abseil-cpp -I../../third_party/boringssl/src/include -I../../third_party/protobuf/src -Igen/protoc_out -I../../third_party/ced/src -I../../third_party/icu/source/common -I../../third_party/icu/source/i18n -I../../third_party/skia -I../../third_party/wuffs/src/release/c -I../../third_party/vulkan/include -I../../third_party/vulkan-deps/vulkan-headers/src/include -fno-delete-null-pointer-checks -fno-ident -fno-strict-aliasing --param=ssp-buffer-size=4 -fstack-protector -fno-unwind-tables -fno-asynchronous-unwind-tables -fPIC -pthread -fcolor-diagnostics -fmerge-all-constants -fcrash-diagnostics-dir=../../tools/clang/crashreports -mllvm -instcombine-lower-dbg-declare=0 -ffp-contract=off -m64 -march=x86-64 -msse3 -Xclang -fdebug-compilation-dir -Xclang . -no-canonical-prefixes -Wall -Wextra -Wimplicit-fallthrough -Wunreachable-code-aggressive -Wthread-safety -Wextra-semi -Wno-missing-field-initializers -Wno-unused-parameter -Wloop-analysis -Wno-unneeded-internal-declaration -Wenum-compare-conditional -Wno-psabi -Wno-ignored-pragma-optimize -Wno-builtin-assume-aligned-alignment -Wno-unused-but-set-parameter -Wno-unused-but-set-variable -O2 -fdata-sections -ffunction-sections -fno-unique-section-names -fno-omit-frame-pointer -g0 -ftrivial-auto-var-init=pattern -fvisibility=hidden -Wheader-hygiene -Wstring-conversion -Wtautological-overlap-compare -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -DPROTOBUF_ALLOW_DEPRECATED=1 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -std=c++14 -fno-trigraphs -Wno-trigraphs -fno-exceptions -fno-rtti -fvisibility-inlines-hidden     -Wno-unknown-warning-option -c ../../components/paint_preview/common/subset_font.cc -o obj/components/paint_preview/common/common/subset_font.o
../../components/paint_preview/common/subset_font.cc:82:40: error: use of undeclared identifier 'HB_SUBSET_SETS_NO_SUBSET_TABLE_TAG'
      hb_subset_input_set(input.get(), HB_SUBSET_SETS_NO_SUBSET_TABLE_TAG);
                                       ^
1 error generated.
ninja: build stopped: subcommand failed.
=> ERROR: ungoogled-chromium-95.0.4638.54_1: do_build: 'ninja -C out/Release ${makejobs} chrome chromedriver mksnapshot chrome_crashpad_handler chrome_sandbox' exited with 1
=> ERROR:   in do_build() at srcpkgs/ungoogled-chromium/template:316

```

I would like to be able to learn and solve something like this in the future. Thanks.

**Comment by DAINRA** [comment]
Well that line comes from [chromium-95-harfbuzz-3.patch](https://github.com/void-linux/void-packages/blob/master/srcpkgs/chromium/patches/chromium-95-harfbuzz-3.patch) but that should not be causing a fail (at least it did not for me when i rebuilt a few days ago for v95.0.4638.69). I left a rebuild running now for v95.0.4638.54 to see if I can reproduce it, but it's gonna take around 1h to compile. In the mean time can you attach an archived copy of your srcpkgs/ungoogled-chromium folder ?

**Comment by Frick-David** [comment]
Let me try rebuilding it. 

Does Github not allow `.tar` file uploads for archives? What would be a good file format for you for the archive? .zip?

**Comment by DAINRA** [comment]
Github docs say you can attach .zip or .gz.
If you are looking for some sort of automation you can take look at what this workflow does [create-release.yml](https://github.com/DAINRA/ungoogled-chromium-void/blob/master/.github/workflows/create-release.yml) (the checkout steps redownload everything) or [testing-ci.yml](https://github.com/DAINRA/ungoogled-chromium-void/blob/master/.github/workflows/testing-ci.yml) (this one doesn't clean void-packages, it just updates). I just use the create release one atm since it's a bit faster to redownload than to pull the changes, tho that depends on internet speed (I got a 1Gbps internet connection and the builder machine is on a 100Mbps line, so not a big deal for me to download the latest void-packages tree + bootstraps + sources).

**Comment by Frick-David** [comment]
Fair enough on the cron job. 
At the very least, I am going to automate the install for a fresh install of void and then have a script to update.

**Comment by Frick-David** [comment]
It is almost finished building. You were correct on pulling for void-pkgs running the bootstrap and cleaning it. Many thinks. I will remember to do that in the future. I was thinking I was missing the harfbuzz library but I saw that Chromium and or void packages gets the dependencies to build that package and they are not necessarily needed to be installed ahead of time. Many thanks. Closing the issue now.
